pkg/snapshot/snapshot_file.go

```diff
@@ -77,18 +94,22 @@ func (md *MilestoneDiff) TreasuryOutput() *utxo.TreasuryOutput {
 		return nil
 	}
 	to := receipt.Transaction.Output
-	msIDPtr, err := md.Milestone.ID()
+	msID, err := md.Milestone.ID()
 	if err != nil {
 		panic(err)
 	}
 	utxoTo := &utxo.TreasuryOutput{Amount: to.Amount}
-	copy(utxoTo.MilestoneID[:], msIDPtr[:])
+	copy(utxoTo.MilestoneID[:], msID[:])
 	return utxoTo
 }
 
 func (md *MilestoneDiff) MarshalBinary() ([]byte, error) {
 	var b bytes.Buffer
 
+	var msDiffLength int64
+	// we increase the offsets here, but we add the bytes at the end because we can not seek in the buffer
+	increaseOffsets(serializer.UInt32ByteSize, &msDiffLength)
```

Comment:
this looks overly complex and error-prone. Better to use `b.Len()` to get the size of the buffer instead of keeping track of the size separately